### PR TITLE
Infrastructure update

### PR DESCRIPTION
**Pull request by vanderhe**
Addresses various infrastructure-related bugs/updates and trivial changes.

### CODE REVIEW

File: cmake/Modules/FindCustomBlas.cmake [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Distributed under the OSI-approved BSD 2-Clause License.
 #
-# Copyright (C) 2021 DFTB+ developers group
+# Copyright (C) 2024 DFTB+ developers group
```
Comment (bhourahine) [inline cmake/Modules/FindCustomBlas.cmake:3]: Unless the file substantially changed, the 2021 date, or possibly a range, is more appropriate.

File: tools/utils/srccheck/pylint/pylintrc-3.ini
Comment (bhourahine) [inline tools/utils/srccheck/pylint/pylintrc-3.ini:1]: Any reason to remove the pylint3 files?

File: cmake/Modules/FindCustomBlas.cmake (hunk 1 above)
Comment (vanderhe) [reply cmake/Modules/FindCustomBlas.cmake:3]: I have reverted the changes, including the `FindCustomLapack.cmake` case.

File: tools/utils/srccheck/pylint/pylintrc-3.ini
Comment (vanderhe) [reply tools/utils/srccheck/pylint/pylintrc-3.ini:1]: Probably outdated, not even the core developers use it, the codebase for sure does not meet any of the criteria, so the default config should do the trick equally well/bad.

Comment (bhourahine) [reply tools/utils/srccheck/pylint/pylintrc-3.ini:1]: OK

**Review by bhourahine — APPROVED** [review]
LGTM

**Pull request merged**